src/trans_rdma.c

```diff
@@ -175,6 +175,48 @@ void __attribute__ ((destructor)) msk_internals_fini(void) {
 	}
 }
 
+/**
+ * Reset global state.
+ * Should (only) be called to reset global data in child process after a fork.
+ */
+void msk_lib_reset(void)
+{
+	/* Close epoll fd's */
+	if (msk_global_state->cm_epollfd != 0)
+		close(msk_global_state->cm_epollfd);
+
+	if (msk_global_state->cq_epollfd != 0)
+		close(msk_global_state->cq_epollfd);
+
+	if (msk_global_state->stats_epollfd != 0)
+		close(msk_global_state->stats_epollfd);
+
+	/* Free worker pool resources */
+	if (msk_global_state->worker_pool.w_efd != 0)
+		close(msk_global_state->worker_pool.w_efd);
+
+	if (msk_global_state->worker_pool.m_efd != 0)
+		close(msk_global_state->worker_pool.m_efd);
+
+	if (msk_global_state->worker_pool.thrids)
+		free(msk_global_state->worker_pool.thrids);
+
+	if (msk_global_state->worker_pool.wd_queue)
+		free(msk_global_state->worker_pool.wd_queue);
```

Comment:
(these are missing in `msk_internals_fini` and should be added there regardless of reusing the functions; thanks for finding these)